browser/ui/webui/brave_new_tab_message_handler.cc

```diff
@@ -78,6 +113,20 @@ BraveNewTabMessageHandler* BraveNewTabMessageHandler::Create(
   // Initial Values
   // Should only contain data that is static
   //
+  auto* ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile);
+  // For safety, default |is_ads_supported_locale_| to true. Better to have
+  // false positive than falsen egative,
+  // in which case we would not show "opt out" toggle.
+  bool is_ads_supported_locale_ = true;
```

Comment:
I think we don't need to query to ads service because manager can know it also?

Reply:
@simonhong the page needs to know if it's a non-supported region so that we don't show the preference to "never show sponsored images". If we just rely on the data manager not returning valid data ever, we won't know if it's just not returning data because there isn't one, or because there cannot be one.

Reply:
Curious why that page should know the difference if there is no data for current region?

Reply:
Ah... we want to show `never show sponsored images` in settings for supported region even if there is no ads campaign?